packages/shared/lib/core/profile/actions/active-profile/login.ts

```diff
@@ -57,6 +65,28 @@ export async function login(loginOptions?: ILoginOptions): Promise<void> {
             }
             setSelectedWallet(initialSelectedWalletId)
 
+            // we need to initialize selectedWallet first and then get the deposit address, otherwise the deposit address will be empty
+            const _selectedWallet = get(selectedWallet)
```

Comment:
the logic of the deposit address should not be in login, we should try to keep it where it was before, in the build wallet logic